tests/components/ipp/test_init.py

```diff
@@ -0,0 +1,42 @@
+"""Tests for the IPP integration."""
+import aiohttp
+
+from homeassistant.components.ipp.const import DOMAIN
+from homeassistant.config_entries import (
+    ENTRY_STATE_LOADED,
+    ENTRY_STATE_NOT_LOADED,
+    ENTRY_STATE_SETUP_RETRY,
+)
+from homeassistant.core import HomeAssistant
+
+from tests.components.ipp import init_integration
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+
+async def test_config_entry_not_ready(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test the IPP configuration entry not ready."""
+    aioclient_mock.post(
+        "http://EPSON123456.local:631/ipp/print", exc=aiohttp.ClientError
+    )
+
+    entry = await init_integration(hass, aioclient_mock)
+    assert entry.state == ENTRY_STATE_SETUP_RETRY
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5274251)